why-polars-for-data-cleaning/index.qmd

```diff
@@ -0,0 +1,185 @@
+-   We want a tool which users will find easy to learn and use.
+-   We do not want to build a tool from scratch.
+
+## Considered options
+
+::: content-hidden
+List and describe some of the options, as well as some of the benefits
+and drawbacks for each option.
+:::
+
+There are a number of tools available to use for data cleaning, but as
+we want something which will run in python, and which is well supported,
+we have narrowed our options down to the following:
+
+-   [DuckDB](https://www.duckdb.org/)
+-   [Pandas](https://pandas.pydata.org/)
+-   [Polars](https://pola.rs/)
+
+### DuckDB
+
+DuckDB is as the name implies not a Python library, but an in-process
+OLAP database solution. It is a very young tool, version 1.0.0 was
+released in 2024, although the first version was released in 2019. The
+developers have set it up as a not-for-profit foundation in the
+Netherlands.
+
+::: columns
+::: column
+#### Benefits
+
+-   Although a very young project it is already in use by a number of
+    large companies and has a large community of users.
+    -   The code is open source, and the way it has been set up is
+        promising for it to remain that way.
+-   It is very fast, and can handle large data sets.
+:::
+
+::: column
+#### Drawbacks
+
+-   It is not designed to be used as a data cleaning tool, but as a
+    database.
+:::
+:::
+
+### Polars
+
+The Polars project started in 2020 and is based in the Netherlands.
+Relatively small team but seems to have a fast growing user base, and
+very enthusiastic developers. Polars is written in Rust and available
+for Python. It is designed to be faster than Pandas, and to handle all
+common data formats. The two main developers are described as
+enthusiastic about open source, which makes it a potentially good choice
+for us.
+
+::: columns
+::: column
+#### Benefits
+
+-   Enthusiastic user community, with a lot of support available.
+-   Supports several languages, including Python.
+-   Handles all common data formats.
+-   Faster than Pandas.
+:::
+
+::: column
+#### Drawbacks
+
+-   Relatively young project, so may not have all the features of more
+    established tools.
+-   Not as well known as Pandas, but certainly getting there.
+-   Compared with Pandas there aren't as many online courses available.
+:::
+:::
+
+### Pandas
+
+Pandas was initially released in 2008, and is now a well established
+tool to work with data frames in Python. The tool is well supported, and
+has a large user base. It is also well documented, and there are a lot
+of online courses available for users to learn how to use it. The tool
+is written in Python, and is designed to be easy to use, and to work
+with large data sets. The main critique is that it can be slow when
+working with large data sets.
+
+::: columns
+::: column
+#### Benefits
+
+-   Established tool with a large user base.
+-   Well supported, and well documented.
+-   Works with all relevant languages and data formats.
```

Comment:
Comments about the syntax and how easy the documentation is to navigate?

Reply:
Don't think I'm at a point where I can judge the syntax, but I've added tutorials and documentation.